include/structures.h

```diff
@@ -0,0 +1,71 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   structures.h                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: dkaiser <dkaiser@student.42heilbronn.de    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/06/25 17:26:40 by dkaiser           #+#    #+#             */
+/*   Updated: 2024/06/25 17:49:43 by dkaiser          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#ifndef STRUCTURES_H
+# define STRUCTURES_H
+
+# include <stdlib.h>
+
+enum							e_token_type
+{
+	PIPE,
+	STRING,
+	REDIRECT_IN,
+	REDIRECT_OUT
+};
+
+enum							e_ast_type
+{
+	PIPELINE,
+	CMD,
+	STRING_LITERAL,
+	FSTRING
+};
+
+typedef struct s_token
+{
+	int							type;
+	char						*content;
+	struct s_token				*previous;
+	struct s_token				*next;
+}								t_token;
+
+typedef struct s_ast_node
+{
+	int							type;
+	union u_ast_node_content	*content;
+}								t_ast_node;
+
+union							u_ast_node_content
```

Comment:
This probably has to be changed. 
If t_cmd can be the content of an ast node, it doesn't make any sense to also have strings.
They would already be incorporated in t_cmd.
This has to be redesigned and restructured.